SharedPackages/DataBrokerProtectionCore/Sources/DataBrokerProtectionCore/Operations/SubJob/BrokerProfileOptOutSubJob.swift

```diff
@@ -0,0 +1,258 @@
+                                           extractedProfileId: Int64,
+                                           brokerId: Int64,
+                                           profileQueryId: Int64,
+                                           database: DataBrokerProtectionRepository,
+                                           notificationCenter: NotificationCenter) {
+        Logger.dataBrokerProtection.log("Finished opt-out operation: \(brokerProfileQueryData.dataBroker.name, privacy: .public)")
+
+        try? database.updateLastRunDate(Date(), brokerId: brokerId, profileQueryId: profileQueryId, extractedProfileId: extractedProfileId)
+        do {
+            try updateOperationDataDates(
+                origin: .optOut,
+                brokerId: brokerId,
+                profileQueryId: profileQueryId,
+                extractedProfileId: extractedProfileId,
+                schedulingConfig: brokerProfileQueryData.dataBroker.schedulingConfig,
+                database: database
+            )
+        } catch {
+            handleOperationError(
+                origin: .optOut,
+                brokerId: brokerId,
+                profileQueryId: profileQueryId,
+                extractedProfileId: extractedProfileId,
+                error: error,
+                database: database,
+                schedulingConfig: brokerProfileQueryData.dataBroker.schedulingConfig
+            )
+        }
+        notificationCenter.post(name: DataBrokerProtectionNotifications.didFinishOptOut, object: brokerProfileQueryData.dataBroker.name)
+    }
+
+    private func incrementOptOutAttemptCountIfNeeded(database: DataBrokerProtectionRepository,
+                                                     brokerId: Int64,
+                                                     profileQueryId: Int64,
+                                                     extractedProfileId: Int64) throws {
+        guard let events = try? database.fetchOptOutHistoryEvents(brokerId: brokerId, profileQueryId: profileQueryId, extractedProfileId: extractedProfileId),
+              events.max(by: { $0.date < $1.date })?.type == .optOutRequested else {
+            return
+        }
+
+        try database.incrementAttemptCount(brokerId: brokerId, profileQueryId: profileQueryId, extractedProfileId: extractedProfileId)
+    }
+
+    private func fetchTotalNumberOfOptOutAttempts(database: DataBrokerProtectionRepository,
+                                                  brokerId: Int64,
+                                                  profileQueryId: Int64,
+                                                  extractedProfileId: Int64) throws -> Int {
+        let events = try database.fetchOptOutHistoryEvents(
+            brokerId: brokerId,
+            profileQueryId: profileQueryId,
+            extractedProfileId: extractedProfileId
+        )
+
+        return events.filter { $0.type == .optOutStarted }.count
+    }
+
+    // MARK: - Generic Job Logic
```

Comment:
This generic job logic is currently duplicated between the two types, and should be extracted out and tested. I'll do this on Monday.